packages/react-hooks/src/ConnectionProvider.tsx

```diff
@@ -15,14 +15,71 @@ import {
   addRecord,
 } from './recordUtils'
 
+export type useConnectionsOptions = {
+  excludedTypes?: [ConnectionType | string]
+  connectionState?: DidExchangeState
+}
+
 const ConnectionContext = createContext<RecordsState<ConnectionRecord> | undefined>(undefined)
 
-export const useConnections = () => {
+/**
+ * This method retreives the connection context for the current agent.
+ * From this you can access all connection records for the agent.
+ * @param options options for useConnections hook, lets us filter out specific types and limit states
+ * @returns a connection context containing information about the agents connections
+ */
+export const useConnections = (options: useConnectionsOptions = {}) => {
   const connectionContext = useContext(ConnectionContext)
   if (!connectionContext) {
     throw new Error('useConnections must be used within a ConnectionContextProvider')
   }
-  return connectionContext
+
+  let connections = connectionContext.records
+
+  // Only run this code if we don't have options.excludedTypes to limit ammount of loops
+  if (options.connectionState && !options.excludedTypes) {
+    const filteredByState = useMemo(
+      () => connections.filter((c: ConnectionRecord) => c.state === options.connectionState),
+      [connections, options.connectionState]
+    )
+    connections = filteredByState
+  }
+
+  if (options.excludedTypes) {
+    const filteredByType = useMemo(
+      () =>
+        connections.filter((record: ConnectionRecord) => {
+          const recordTypes = (record.getTag('connectionType') as [string] | null) || []
```

Comment:
Is it easier to just do:

```js
const type = (r.getTag('connectionType') || []) as [string]
```

Since you are short circuiting to an empty array anyway?